File: frontend/src/lib/types/alerts.ts
// lib/types/alerts.ts
export interface AlertBase {
    type: string;
    niveau: 'info' | 'warning' | 'critical';
    message: string;
    espece?: string;
    animal_id?: number;
    enclos_id?: number;
    entite_type?: string;
    entite_id?: number;
    date_alerte: string;
    date_limite?: string;
}

export interface AlertCreate extends AlertBase { }
export interface AlertResponse extends AlertBase {
    id: number;
    est_lue: boolean;
    est_traitee: boolean;
    utilisateur_traitement_id?: number;
    date_traitement?: string;
    created_at: string;
}

File: backend/app/schemas/alerts.py
# backend/app/schemas/alerts.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class AlertBase(BaseModel):
    type: str  # vaccination, pesee, nettoyage, mortalite, water_quality, bea, comptable
    niveau: str  # info, warning, critical
    message: str
    espece: Optional[str] = None
    animal_id: Optional[int] = None
    enclos_id: Optional[int] = None
    entite_type: Optional[str] = None
    entite_id: Optional[int] = None
    date_alerte: datetime
    date_limite: Optional[datetime] = None


class AlertCreate(AlertBase):
    pass


class AlertResponse(AlertBase):
    id: int
    est_lue: bool
    est_traitee: bool
    utilisateur_traitement_id: Optional[int]
    date_traitement: Optional[datetime]
    created_at: datetime
    
    class Config:
        from_attributes = True